### PR TITLE
*: refactor log of dmctl

### PR DESCRIPTION
**Pull request by GregoryIan**
### What problem does this PR solve? 
refactor log of dmctl

### What is changed and how it works?
use pingcap/log repo

### CODE REVIEW

File: pkg/log/log.go [hunk 1]
```diff
@@ -14,196 +14,111 @@
...
-			if k != "file" && k != "line" {
-				fmt.Fprintf(b, " %v=%v", k, v)
-			}
-		}
-	}
-
-	b.WriteByte('\n')
-
-	return b.Bytes(), nil
+// L returns the current logger for Lightning.
```
Comment (kennytm) [inline pkg/log/log.go]: 🙃

**Comment by csuzhangxc** [comment]
@GregoryIan do not need to care test cases failed caused by _log_ now?

**Comment by GregoryIan** [comment]
@csuzhangxc no need, it must fail

File: pkg/log/log.go (hunk 1 above)
Comment (GregoryIan) [reply pkg/log/log.go]: really don't want to import lightning repo, it would be nice if It was in a public library.

Comment (kennytm) [reply pkg/log/log.go]: Yes we could copy the common parts into tidb-tools

File: checker/checker.go [hunk 2]
```diff
@@ -289,15 +294,15 @@ func (c *Checker) closeDBs() {
...
+		c.logger.Warn("try to pause, but already closed")
 		return
 	}
 }
 
 // Resume resumes the paused process
 func (c *Checker) Resume(ctx context.Context, pr chan pb.ProcessResult) {
 	if c.closed.Get() {
-		log.Warn("[checker] try to resume, but already closed")
+		c.logger.Warn("[checker] try to resume, but already closed")
```
Comment (csuzhangxc) [inline checker/checker.go]:
```suggestion
		c.logger.Warn("try to resume, but already closed")
```

File: cmd/dm-ctl/main.go [hunk 3]
```diff
@@ -48,8 +48,14 @@ func main() {
 
 	// now, we use checker in dmctl while it using some pkg which log some thing when running
 	// to make dmctl output more clear, simply redirect log to file rather output to stdout
-	log.SetLevelByString("info")
-	log.SetOutputByName("dmctl.log")
+	err = log.InitLogger(&log.Config{
+		File:  "dmctl.log",
```
Comment (csuzhangxc) [inline cmd/dm-ctl/main.go]: is it needed to _Sync_ log to file?

File: checker/checker.go [hunk 4]
```diff
@@ -23,6 +23,8 @@ import (
 	"sync"
 	"time"
 
+	"go.uber.org/zap"
+
```
Comment (csuzhangxc) [inline checker/checker.go]: no extra blank line needed? ditto for other files.

File: pkg/parser/common.go [hunk 5]
```diff
@@ -38,13 +40,12 @@ var (
 // Parse wraps parser.Parse(), makes `parser` suitable for dm
 func Parse(p *parser.Parser, sql, charset, collation string) (stmt []ast.StmtNode, err error) {
 	stmts, warnings, err := p.Parse(sql, charset, collation)
-
-	for _, warning := range warnings {
-		log.Warnf("parsing sql %s:%v", sql, warning)
+	if err != nil {
+		log.L().Error("parse statement", zap.String("sql", sql), zap.Reflect("error message", err))
```
Comment (csuzhangxc) [inline pkg/parser/common.go]: can we use `zap.Error(err)` or `log.ShortError(err)` directly here?

File: pkg/utils/common.go [hunk 6]
```diff
@@ -179,27 +181,27 @@ func querySQL(db *sql.DB, query string, maxRetry int) (*sql.Rows, error) {
...
+			log.L().Warn("query retry", zap.String("query", query), log.ShortError(err))
 			continue
 		}
 
 		return rows, nil
 	}
 
 	if err != nil {
-		log.Errorf("query sql[%s] failed %v", query, errors.ErrorStack(err))
+		log.L().Error("query retry", zap.String("query", query), zap.Error(err))
```
Comment (csuzhangxc) [inline pkg/utils/common.go]: is `log.ShortError(err)` better?

Comment (GregoryIan) [reply pkg/utils/common.go]: as old log `errors.ErrorStack`, I use `zap.Error`

File: pkg/parser/common.go (hunk 5 above)
Comment (GregoryIan) [reply pkg/parser/common.go]: yep we can, I just want to make it same with warning messages

File: cmd/dm-ctl/main.go (hunk 3 above)
Comment (GregoryIan) [reply cmd/dm-ctl/main.go]: I  thought it,  I even think we don't need log here, so I don't add it

**Comment by GregoryIan** [comment]
@csuzhangxc PTAL again

File: pkg/utils/printer.go [hunk 7]
```diff
@@ -16,10 +16,9 @@ package utils
 import (
 	"fmt"
 
+	"github.com/pingcap/dm/pkg/log"
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
-
-	"github.com/pingcap/dm/pkg/log"
```
Comment (csuzhangxc) [inline pkg/utils/printer.go]: we can keep this _import_ in a separate group here.

File: pkg/utils/common.go [hunk 8]
```diff
@@ -179,27 +178,27 @@ func querySQL(db *sql.DB, query string, maxRetry int) (*sql.Rows, error) {
...
+			log.L().Warn("query retry", zap.String("query", query), log.ShortError(err))
 			continue
 		}
 
 		return rows, nil
 	}
 
 	if err != nil {
-		log.Errorf("query sql[%s] failed %v", query, errors.ErrorStack(err))
+		log.L().Error("query retry", zap.String("query", query), log.ShortError(err))
```
Comment (amyangfei) [inline pkg/utils/common.go]: here is query failed at last, no more retry.
btw the logic is redundant, the `err` is always not nil in L196

**Pull request merged**